### ISSUE TITLE
Loading a Node without width/height/viewBox results in layer escaping view bounds

### ISSUE THREAD

**Issue by rafaelclaycon** [exyte/SVGView#35]
We're trying to separate an SVG's layers to be able to control each separately, so we implemented a method like this:

```
func node(for layer: MapLayers) -> SVGNode {
    return rootNode.getNode(byId: layer.rawValue) ?? .init()
}
```

The problem is loading layers like this results in the layer "escaping" the bounds of our view. Something told me it had to do with the viewBox part that now was "missing" (not included since we're not loading the entire SVG anymore).

So, to be clear, what works:

- Loading an entire SVG, with either width/height OR viewBox in the svg tag.

What doesn't:

- Loading a single layer gotten through the above method.
- Loading an entire SVG that had width/height **and** viewBox removed in the svg tag.

To fix this, is there a way for me to specify width, height and viewBox for each Node (I tried .frame(), that didn't work) or maybe inject this svg tag into each layer?

**Comment by vcsoares** [comment]
I've managed to find a workaround for this.

**TL;DR**: `SVGViewport` should be made public, as it seems to be responsible for resizing and repositioning child nodes according to the specified viewBox. Most of its methods are already marked as public, but the class itself is not - which leads me to think this was an oversight.

While we're unable to instantiate a new Viewport directly, what we can do is manually fetch the parent's view bounds with a `GeometryReader`, for example, and then apply a transform by instantiating a `SVGPreserveAspectRatio` object and calling its `layout(size:into:)` method.

**Comment by f3dm76** [comment]
Looks like you figured it out. And SVGViewport is now public. please feel free to reopen if you are still experiencing issues with this.
